src/Hint/Bracket.hs

```diff
@@ -190,10 +190,14 @@ bracket pretty isPartialAtom root = f Nothing
     -- 'x' actually need bracketing in this context?
     f (Just (i, o, gen)) v@(remParens' -> Just x)
       | not $ needBracket i o x, not $ isPartialAtom (Just o) x =
-          rawIdea Suggestion msg (getLoc v) (pretty o) (Just (pretty (gen x))) [] [r] : g x
+          rawIdea Suggestion msg (getLoc v) from to [] [r] : g x
       where
         typ = findType v
         r = Replace typ (toSS v) [("x", toSS x)] "x"
+        (from, to) = reduceJunks (pretty o) (Just (pretty (gen x)))
+        reduceJunks :: String -> Maybe String -> (String, Maybe String)
```

Comment:
Pattern match is not exhaustive. On line 193, you should change `to` to `Just to` so that `reduceJunks` become `String -> String -> (String, String)`.

Reply:
> Please give a before-after comparison.

```haskell
main = do
    let a = 1
    let a = 2
    let a = 3
    let a = 4
    let a = (f 7)
    let a = 8
    let a = 9
    let a = 10
    return ()   
  ```
Before:
```
Found:
  do let a = 1    
     let a = 2    
     let a = 3    
     let a = 4    
     let a = 5    
     let a = 6    
     let a = (f 7)
     let a = 8
     let a = 9
     let a = 10
     return ()
Perhaps:
  do let a = 1
     let a = 2
     let a = 3
     let a = 4
     let a = 5
     let a = 6
     let a = f 7
     let a = 8
     let a = 9
     let a = 10
     return ()
```
After:
```
Found:
     let a = (f 7)

Perhaps:
     let a = f 7
 ```